File: 125-valid-palindrome/125-valid-palindrome.scala
object Solution {
    def isPalindrome(s: String): Boolean = {
        var i = 0
        var j = s.length - 1
        var res = true
        
        while(i < j) {
            while (i < j && !s.charAt(i).isLetter && !s.charAt(i).isDigit) { i += 1 }
            while (i < j && !s.charAt(j).isLetter && !s.charAt(j).isDigit) { j -= 1 }
            
            res &&= s.charAt(i).toLower == s.charAt(j).toLower
            
            i += 1
            j -= 1
        }
        
        res
    }
}

File: 125-valid-palindrome/125-valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    var isLetter = function(c) {
        return /[a-zA-Z0-9]/.test(c);
    }
    
    let left = 0;
    let right = s.length - 1;
    
    while(left < right) {
        while(left < right && !isLetter(s.charAt(left))) left++;
        while(left < right && !isLetter(s.charAt(right))) right--;
        
        if (left < right && s.charAt(left).toLowerCase() != s.charAt(right).toLowerCase()) return false;
        
        left++;
        right--;
    }
    
    return true;
};